### PR TITLE
Mongodb schema ln

### PR DESCRIPTION
**Pull request by lavnimusha**
### What this PR does (required):
- Added MongoDB schema and profile controllers

### Screenshots / Videos (required):
Image 8-15-21 at 10 51 PM
Image 8-15-21 at 11 01 PM
Image 8-15-21 at 11 07 PM
Image 8-16-21 at 12 59 AM

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/db.js [hunk 1]
```diff
@@ -10,5 +10,14 @@ const connectDB = async () => {
...
+		FirstName: String,
+		LastName: String,
+		Description: String,
+		Availability: 	Date
 
-module.exports = connectDB;
+	 });
+const Profile = mongoose.model('Profile', ProfileSchema);
+
+module.exports = { connectDB, Profile };
```
Comment (ugur-akin) [inline server/db.js:23]: Might be a good idea to extract models in their own file/folder, otherwise looks good!

File: server/db.js [hunk 2]
```diff
@@ -10,5 +10,14 @@ const connectDB = async () => {
 
   console.log(`MongoDB Connected: ${conn.connection.host}`.cyan.underline.bold);
 };
+const Schema  = mongoose.Schema;
+const ProfileSchema =  new Schema({
```
Comment (rajivtitus) [inline server/db.js:14]: As mongoose is a JavaScript based Data Modeling Language, I would recommend the usage of camelCase for field/variable names.

File: server/db.js [hunk 3]
```diff
@@ -10,5 +10,14 @@ const connectDB = async () => {
 
   console.log(`MongoDB Connected: ${conn.connection.host}`.cyan.underline.bold);
 };
+const Schema  = mongoose.Schema;
+const ProfileSchema =  new Schema({ 
+		FirstName: String,
+		LastName: String,
+		Description: String,
+		Availability: 	Date
```
Comment (rajivtitus) [inline server/db.js:18]: Are we planning to have a date range (start-end) or availability throughout the (Mon-Fri) week? :)
(a single date won't fit our use case for availability)

File: server/db.js (hunk 1 above)
Comment (lavnimusha) [reply server/db.js]: Resolved.

**Review by rajivtitus — APPROVED** [review]
The model LGTM! Thanks for making request changes.

Note: For future PRs please update the base of the PR, as the Profile Controllers are in a separate PR but the same code in also visible in this PR. Thank you!

File: server/routes/profile.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { searchProfiles, createProfile, listAllProfiles, updateProfile } = require("../controllers/profile");
+
+router.route("/create").post(createProfile);
```
Comment (rajivtitus) [inline server/routes/profile.js:6]: Maybe add the protect middleware to this route as well?

File: server/controllers/profile.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/generateToken");
+
+// @route POST /profiles
+// @desc Given parameters passed in create a profile
+// @access Private
+
+ exports.createProfile = asyncHandler(async (req, res, next) => {
+
+    const { firstName, lastName, description, email, phoneNumber, availability:{startDate,endDate,daysOfWeek} } = req.query;
```
Comment (rajivtitus) [inline server/controllers/profile.js:11]: Why are we using query parameters, shouldn't we be getting these fields from the request body (req.body) ?
Ditto throughout!

File: server/controllers/profile.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        res.cookie("token", token, {
+          httpOnly: true,
+          maxAge: secondsInWeek * 1000
+        });
+    
+        res.status(201).json({
+          success: {
+            user: {
+              id: profile._id,
+              ffirstName: profile.firstName,
```
Comment (rajivtitus) [inline server/controllers/profile.js:39]: Typo here :)

File: server/controllers/profile.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+// @route POST /profiles
+// @desc Update profile for given id
+// @access Private
+
+exports.updateProfile = asyncHandler( async(req, res, next) => {
+  
+  const { firstName, lastName, description, email, phoneNumber } = req.query;
+  const profile_id = req.query._id;
+
+    const profile = Profile.findOne({_id: profile_id}, function(err) {
```
Comment (rajivtitus) [inline server/controllers/profile.js:108]: You can look in the usage of mongoDB 'findById&Update'

**Pull request merged**